src/libstd/sys/unix/fs.rs

```diff
@@ -132,6 +139,32 @@ impl FromInner<raw::mode_t> for FilePermissions {
 impl Iterator for ReadDir {
     type Item = io::Result<DirEntry>;
 
+    #[cfg(target_os = "sunos")]
```

Comment:
Hm. Well, it looks like there is a standard-compliant `readdir_r`, actually, though it seems to be linked in some special way. I'll try to figure it out, thanks!